### PR TITLE
feat: Rename: Datadog -> DatadogLambda

### PR DESCRIPTION
**Pull request by lym953**
### What does this PR do?

1. Rename `Datadog` class as `DatadogLambda`, including:
    1. rename `datadog.ts` as `datadog-lambda.ts`
    2. rename `datadog.spec.ts` as `datadog-lambda.spec.ts` 
3. Make all references of `Datadog` use `DatadogLambda`
4. Create a new `Datadog` class which just wraps (extends) the new `DatadogLambda` class, so its behavior will be the same as the old `Datadog` class. Let me know if you think I should wrap it in a different way.
5. Create an example stack `lambda-function-stack-legacy-datadog-api.ts`, which uses `Datadog` instead of `DatadogLambda`, showing how to use the legacy `Datadog` API
6. Create a test `datadog.spec.ts` to ensure the `Datadog` API still works

### Motivation

To make it cleaner to add `DatadogStepFunction` class later. Details in [[RFC] Changing API for Datadog CDK Construct](https://docs.google.com/document/d/1g3bFuhWUayvuS_aS6kOHe_CGECocd8IhoKmM9s4dPU0)

#### Automated testing
1. pass existing automated tests
2. pass the new `datadog.spec.ts`
##### 1. using `DatadogLambda` API
Steps:
1. In `examples/typescript-stack/lib/cdk-typescript-stack.ts`, replace the `CdkTypeScriptStack` class with the `ExampleStack` class in `integration_tests/stacks/lambda-function-stack.ts`. I can't use the existing `CdkTypeScriptStack` class due to some docker permission issue.
    - The `ExampleStack` class uses `DatadogLambda` API
2. Follow the instructions in [Datadog CDK Constructs internal wiki](https://datadoghq.atlassian.net/wiki/spaces/SLS/pages/2723906703/Datadog+CDK+Constructs#Testing) to install the local package deploy the stack

Result:
- Deployment is successful. The Lambda functions have Datadog layers installed.

##### 2. using `Datadog` API
Step:
- In the test above, make the `ExampleStack` class use `Datadog` class instead of `DatadogLambda`. Deploy again.

Result:
- Deployment is also successful. The Lambda functions have Datadog layers installed.

### Additional Notes

Next steps:
1. update `README.md`, recommending users to use `DatadogLambda` instead of `Datadog`

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [x] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: integration_tests/stacks/lambda-function-stack-legacy-datadog-api.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/*
```
Comment (lym953) [inline integration_tests/stacks/lambda-function-stack-legacy-datadog-api.ts:1]: Copied and modified from `integration_tests/stacks/lambda-function-stack.ts`

File: src/datadog-lambda.ts [hunk 2]
```diff
@@ -0,0 +1,365 @@
+/*
```
Comment (lym953) [inline src/datadog-lambda.ts:1]: This file is renamed from `datadog.ts`

File: test/datadog-lambda.spec.ts [hunk 3]
```diff
@@ -0,0 +1,687 @@
+import { App, Stack, Token } from "aws-cdk-lib";
+import { Template } from "aws-cdk-lib/assertions";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import { LogGroup } from "aws-cdk-lib/aws-logs";
+import { addCdkConstructVersionTag, checkForMultipleApiKeys, DatadogLambda, DD_HANDLER_ENV_VAR } from "../src/index";
+const { ISecret } = require("aws-cdk-lib/aws-secretsmanager");
+const versionJson = require("../version.json");
```
[the remark this answers is not in the document]
Comment (lym953) [reply test/datadog-lambda.spec.ts:7]: These are just comments on existing code. Let me address them in a separate PR.

File: test/datadog.spec.ts [hunk 4]
```diff
@@ -1,537 +1,11 @@
-import { App, Stack, Token } from "aws-cdk-lib";
-import { Template } from "aws-cdk-lib/assertions";
+import { App, Stack } from "aws-cdk-lib";
```
Comment (lym953) [inline test/datadog.spec.ts:1]: This file was first renamed as `datadog-lambda.spec.ts`. Then `datadog.spec.ts` was created again with one test, to ensure the `Datadog` API still works.

**Review by purple4reina — APPROVED** [review]

**Comment by pwrmiller** [comment]
Just an FYI, this is a breaking change. Packages that depend on datadog-cdk-constructs will now fail, with errors similar to (in Python, for example)

```
ImportError: cannot import name 'DatadogProps' from 'datadog_cdk_constructs_v2'
```

**Pull request merged**